tools/config/__init__.py

```diff
@@ -600,8 +601,13 @@ def has_ram_regions(self):
 
     def deliver_into(self):
         if self.target.deliver_to_target:
-            label_dir = "TARGET_{}".format(self.target.deliver_to_target)
-            target_delivery_dir = join(DELIVERY_DIR, label_dir)
+            delivery_target = Target.get_target(self.target.deliver_to_target)
+            if hasattr(delivery_target, "delivery_dir"):
+                target_delivery_dir = join(ROOT, delivery_target.delivery_dir)
```

Comment:
I don't like that `delivery_dir` is always relative to the root of Mbed OS. This:
 1) Won't work if we ever want to do delivery in the online compiler (which is questionable)
 2) Works different from all of the other path-using configuration variables.

Instead, could we make this relative to the location that defines it?

Reply:
@theotherjimmy Do you mean that if it is defined in `target.json`, it will be relative to the target which defines it?

Reply:
@mikisch81 I don't know what that means. I'm saying that the path will be relative to the file that defines the path.

Reply:
@theotherjimmy
Our use cases are targets (like sequana_m0 and musca_a1_s) which would put the artifacts in their non-secure counterparts (sequana_m4 and musca_a1_ns).

Reply:
@mikisch81 That does not help me understand what
> it will be relative to the target which defines it

means. Targets are all defined in the same spot in the filesystem, so I'm not sure how you could make a path relative to a target and have that be meaningful.

-----------------


I don't think my suggestions change your ability to deliver artifacts from secure target to not-secure target.